### DISCUSSION TITLE
AMI Region Options

### DISCUSSION THREAD

**Discussion by knechtionscoding** [renovatebot/renovate#23854] [Suggest an Idea]
### Tell us more.

In the current implementation Renovate only looks for AMI IDs in the same region as Renovate is running.

This is problematic because AMI IDs differ between regions. Renovate should accept a region parameter alongside the name and owner to tell it where to look up the AMI ID.

This could extend further than just region, basically allowing for any aws sdk variable to be passed in (profile for example).

Considerations:
- This adds more calls to the AWS API
- Adds more information that needs to be cached
- Have to store multiple dependencies with the same name

Happy to submit an initial PR to start the conversation if this is something renovate wants to support

**Comment by knechtionscoding** [comment]
I think we just need to extend: https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/aws-machine-image/index.ts#L38 to allow for region to be passed in as part of the comment?

**Reply by knechtionscoding** [reply]
I'm not sure where the parsing for the comment happens however to add another field for region/profile.

**Reply by knechtionscoding** [reply]
@rarkins can you perhaps point me at how the regex from the manager makes it into the datasource? So I can pass region correctly?

**Reply by ivankatliarchuk** [reply]
Hi. 
Related issue https://github.com/renovatebot/renovate/issues/14421. Currently it reads environment variable e.g. `AWS_REGION` or `AWS_DEFAULT_REGION`

**Comment by reegnz** [comment]
Support just got released: https://github.com/renovatebot/renovate/issues/14421#issuecomment-1697281586
